exec/command_resolver.go

```diff
@@ -31,7 +33,9 @@ type CmdResolver struct {
 }
 
 // NewCmdResolver creates new instance CmdResolver.
-func NewCmdResolver(shellDetector shell.ContainerShellDetector, infoExecCreator exec_info.InfoExecCreator) *CmdResolver {
+func NewCmdResolver(k8sAPI *client.K8sAPI, nameSpace string) *CmdResolver {
+	shellDetector := shell.NewShellDetector(k8sAPI, nameSpace)
+	infoExecCreator := exec_info.NewKubernetesInfoExecCreator(nameSpace, k8sAPI.GetClient().Core(), k8sAPI.GetConfig())
```

Comment:
```suggestion
	infoExecCreator := exec_info.NewKubernetesInfoExecCreator(namespace, k8sAPI.GetClient().Core(), k8sAPI.GetConfig())
```

Reply:
Also, I wonder if it makes sense to propagate K8sAPI wherever it's possible in che-machine-exec to void decoupling client and config?